xrpl-wasm-std/src/core/types/amountv2.rs

```diff
@@ -0,0 +1,323 @@
+use crate::core::types::account_id::AccountID;
+use crate::core::types::asset::Asset;
+use crate::core::types::currency::Currency;
+use crate::core::types::mpt_id::MptId;
+
+/// Represents an amount of value with an associated asset.
+/// This is the Rust equivalent of C++'s STAmount.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[repr(C)]
+pub struct Amount {
+    /// The asset (currency type)
+    asset: Asset,
+    /// The mantissa (value)
+    mantissa: i64,
+    /// The exponent (for IOUs, indicates decimal place)
+    exponent: i32,
+    /// Whether this is negative
+    negative: bool,
+}
+
+impl Amount {
+    /// Create a new XRP amount (in drops)
+    pub const fn xrp(drops: i64) -> Self {
+        let (mantissa, negative) = if drops < 0 {
+            (-drops, true)
+        } else {
+            (drops, false)
+        };
+        
+        Self {
+            asset: Asset::xrp(),
+            mantissa,
+            exponent: 0,
+            negative,
+        }
+    }
+
+    /// Create a new IOU amount
+    pub const fn iou(mantissa: i64, exponent: i32, currency: Currency, account: AccountID) -> Self {
```

Comment:
Making a note to potentially revisit this pending the discussion near `to_bytes`

(at present IOUs shouldn't really be constructed this way, but should instead use the `opaque_float` design).